Update collaboration and teamwork article (#38354)
mdn/content: files/en-us/learn_web_development/getting_started/soft_skills/collaboration_and_teamwork/index.md

  </tbody>
</table>

## Working with others

> As a professional in the web industry, you'll have to spend a lot of time working with your peers, whether you are employed or in education. Your colleagues or fellow students could be a hugely valuable aid to your learning, as we first discussed back in [Research and learning](/en-US/docs/Learn_web_development/Getting_started/Soft_skills/Research_and_learning#build_a_network_of_contacts), and you should at least make sure that you get on with them to begin with.

Reviewer: "whether you are employed or in education." - because educators are unemployed?

Author: Good point. Updated to

> As a web developer, you'll have to spend a lot of time working with your peers, whether you are employed or studying.

> The basic essence of this is to treat them how you would want to be treated. In all engagements, stay polite, be respectful, be coperative, and show empathy for their feelings and needs.

Reviewer: "The basic essence of this is to treat them how you would want to be treated. " - I made this point to someone recently and they said, that assumes that they want to be treated the same way as you - often they are different enough that is not helpful. 

Up to you, but I think this sentence could be removed. Also there is a typo in second sentence.

```suggestion
What's important is to be polite, respectful, cooperative, and show empathy for their feelings and needs.
```

## Help others, ask questions

> If a colleague is struggling, offer to help them. web developers who provide help are much more likely to build strong relationships and get help in return.

Reviewer: ```suggestion
If a colleague is struggling, offer to help them. Web developers who provide help are much more likely to build strong relationships and get help in return.
```

Author: Fixed, ta!

Peer reviews are also a good place to get practice on this ([see below](#teamwork_and_peer_reviews)).

## Publish your work

As a general rule, it is very worthwhile to publish your work, including source code, lessons learned, what was challenging, etc. Your peers will find this very useful, and so will the wider community.

When you find a useful answer, write about it. Put your code on a social coding site like [GitHub](https://github.com/) or [GitLab](https://about.gitlab.com/), and write up your thoughts on a blog. Share your publications via social media.

Not only will the process of writing clarify the concepts to you, but you'll also get validation and/or feedback from the community. It could help you to start making a name for yourself in the industry.

Even if you think your achievements are trivial, you probably know more than you think you do, and there will always be someone at an earlier stage in their journey towards mastering code that will find your work useful.

## Participate in open source

> Another way to make friends and influence people is to find open source projects and offer updates to fix issues in other people's work. Provided you are polite and respectful, the project owners will be very grateful for your contributions, and this will give you valuable experience and practice. For example, [MDN's source code](https://github.com/mdn/content) is available on GitHub, and we have many [issues](https://github.com/mdn/content/issues) that you could work on.

Reviewer: ```suggestion
Another way to make friends and influence people is to find open source projects and offer updates to fix issues in other people's work. Provided you are polite and respectful, and provide your reasoning for changes, the project owners will be very grateful for your contributions, and this will give you valuable experience and practice. For example, [MDN's source code](https://github.com/mdn/content) is available on GitHub, and we have many [issues](https://github.com/mdn/content/issues) that you could work on.
```

Author: Good call; I've fixed this in a slightly different way. I've actually restructured this little section quite a bit.

  </tbody>
</table>

## Working with others

As a web developer, you'll have to spend a lot of time working with your peers, whether you are employed or studying. Your colleagues or fellow students could be a hugely valuable aid to your learning, as we first discussed back in [Research and learning](/en-US/docs/Learn_web_development/Getting_started/Soft_skills/Research_and_learning#build_a_network_of_contacts), and you should at least make sure that you get on with them to begin with.

It is important to treat people with the same level of empathy and respect as you'd wish for. Different people have different requirements emotionally, socially, etc., but at a base level you should be polite, respectful, and cooperative in all engagements.

> [!NOTE]

> > Most organizations publish participation guidelines that define their expectations of behavior for people who participate in their work/activities. These guidelines should also include a definition of what is considered unacceptable behavior, the consequences of unacceptable behavior, and how to report unacceptable behavior. For example, MDN is governed by the wider [Mozilla Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/).

Reviewer 2: ```suggestion
> Most organizations publish a **Code of Conduct** that defines their expectations of behavior for people who participate in their project. These guidelines usually include a definition of what is considered unacceptable behavior, the consequences of unacceptable behavior, and how to report unacceptable behavior. For example, MDN is governed by the wider [Mozilla Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/).
```

Reviewer 2: I'd just mention Code of Conduct or "CoC" so people get used to seeing that or looking out for it.

What's also interesting to look out for is:

> A lot of projects add their guidelines in a "Code of Conduct" document named `CODE_OF_CONDUCT.md`.
> If it's a project that's hosted on GitHub, you should be able to see this as part of the main project metadata, like the [mdn/content README](https://github.com/mdn/content?tab=coc-ov-file).

When you find a useful answer, write about it. Put your code on a social coding site like [GitHub](https://github.com/) or [GitLab](https://about.gitlab.com/), and write up your thoughts on a blog. Share your publications via social media.

Not only will the process of writing clarify the concepts to you, but you'll also get validation and/or feedback from the community. It could help you to start making a name for yourself in the industry.

Even if you think your achievements are trivial, you probably know more than you think you do, and there will always be someone at an earlier stage in their journey towards mastering code that will find your work useful.

## Participate in open source

Another way to make friends and influence people is to find open source projects and offer updates to fix issues in other people's work. Provided you are polite and respectful, and explain your changes and the reasoning behind them, the project owners will be very grateful for your contributions. Your updates will give you valuable experience and practice.

> For example, [MDN's source code](https://github.com/mdn/content) is available on GitHub, and we have many [issues](https://github.com/mdn/content/issues) that you could work on.

Reviewer 2: ```suggestion
For example, [MDN's source code](https://github.com/mdn/content) is available on GitHub, and we have many [issues](https://github.com/mdn/content/issues) that you could work on.
There are plenty of [Community resources](/en-US/docs/MDN/Community) on MDN that describe the project and what contributing to an open source project looks like.
```

- If you are reviewing someone else's work, aim to provide a balance between encouragement and praise for good work, and critical feedback. Practice how to deliver that feedback constructively and respectfully. Even if they seem somewhat trivial, always come up with one or two questions to ask when discussing or reviewing peer work.
- If you work is being reviewed, try not to take criticism personally, and focus on the positives and what you can learn. Assume the best of intentions from your reviewer — they are just trying to help.

## Learn to embrace failure

A very common issue that causes students and new developers to shy away from experimentation and taking risks (for example when starting new projects or exploring new ideas) is fear of failure, and how it will be judged by peers/team mates. You should spend some time learning about the value that can be gleaned from making mistakes.

Failure is really only a small step on the road to success. All of the most successful developers have failed many times, and it is not the actual failure that matters, but how you handle that failure, and the lessons you learn from it.

Here are some tips:

> - Define a safe space/peer group where people are free to ask questions — remember that there are no "stupid questions", just fools who do not question. Foster a culture where peers support each other and failure is not judged harshly.

Reviewer 2: Maybe we can reword it a little :)

```suggestion
- Define a safe space/peer group where people are free to ask questions — remember that there are no "stupid questions", only people who fail to question. Foster a culture where peers support each other and failure is not judged harshly.
```
